packages/test/snapshots/README.md

```diff
@@ -1,6 +1,103 @@
+- Use the [fetch-tool](../../tools/fetch-tool/README.md) tool to fetch and save the snapshots and ops for the document.
+- Create a folder for the new snapshots under [snapshotTestContent](./content/snapshotTestContent).
+- Add the `message.json` file from the output of the [fetch-tool](../../tools/fetch-tool/README.md) to the new folder.
+- Create a sub-folder called `base_snapshot`. From the output of the [fetch-tool](../../tools/fetch-tool/README.md), add the contents of the oldest snapshot (except `tree-original.json` and `decoded` folder).
+  - The downloaded snapshots are in folders whose names are in format `<number>-<versionId>`. The oldest snapshot is in a folder that starts with the highest `number`.
+  - Following is an example output of [fetch-tool](../../tools/fetch-tool/README.md):
+    ```
+    C:\snapshots>dir
+    07/26/2021  05:28 PM    <DIR>          .
+    08/02/2021  01:19 PM    <DIR>          ..
+    07/26/2021  05:28 PM    <DIR>          0-bB8gBAAAHAAAA
+    07/26/2021  05:28 PM    <DIR>          1-bBwAAAAAHAAAA
+    07/26/2021  05:28 PM               680 info.json
+    07/26/2021  05:28 PM           641,806 messages.json
+                2 File(s)        642,486 bytes
+                4 Dir(s)  243,849,199,616 bytes free
+    ```
+    There are two snapshots in folders `0-bB8gBAAAHAAAA` and `1-bBwAAAAAHAAAA`. The oldest snapshot folder is `1-bBwAAAAAHAAAA` whose contents are:
+    ```
+    C:\snapshots\1-bBwAAAAAHAAAA>dir
+    07/26/2021  05:28 PM    <DIR>          .
+    07/26/2021  05:28 PM    <DIR>          ..
+    07/26/2021  05:28 PM                40 0-bARBrcuSOLdT_lXwAHWYYa4cJ
+    07/26/2021  05:28 PM                74 1-bARAWgNwwMAtEWFX6f8eXjarD
+    07/26/2021  05:28 PM               146 3-bARC8xAwm8aYZEMbE2vbPYiTA
+    07/26/2021  05:28 PM               104 4-bARDE9u7Ztbm4_xyTBb9dmJKd
+    07/26/2021  05:28 PM               242 5-bARBgXKoar1aPt2TC8hgpvFzw
+    07/26/2021  05:28 PM               130 6-bARBTxqdFc58itZIePNAhBxVd
+    07/26/2021  05:28 PM                83 7-bARCBnzMy4FNE7kexVfsvTEs0
+    07/26/2021  05:28 PM               183 8-bARCZTcvC68fmXTMjacLU6Tct
+    07/26/2021  05:28 PM               351 9-bARBk1cbwDcU8wOEeGmmuJxxh
+    07/26/2021  05:28 PM    <DIR>          decoded
+    07/26/2021  05:28 PM             2,231 tree-original.json
+    07/26/2021  05:28 PM             2,308 tree.json
+                27 File(s)          9,229 bytes
+                3 Dir(s)  243,653,701,632 bytes free
+    ```
+    Copy everything in this folder except `tree-original.json` and `decode` to the `base_snapshot` directory created above.
+- Run the following from the `packages\test\snapshots` directory:
+  ```
+  npm run test:new
+  ```
+  This will generate the snapshots for this based on the current version of the runtime. These new snapshots will be under `current_snapshots` directory.
+- Run the tests to validate that all the tests pass with the new snapshots.
```

Comment:
Is there a playbook about what thread to follow if the tests do not pass with snapshots? I saw some failure messages from these tests and they seem rather cryptic (like a windiff telling you that lines don't match).

Reply:
That's because it is what the tests do. All the test cases generate snapshots in two clients at the same point in time (sequence number) and compare the snapshots. The variations are how the clients got to that point. If the snapshot comparison fails, it spits out the section of the snapshot that failed to compare.

Reply:
Debugging these tests are rather hard when something fails. There are couple of scenarios where the test say that - Looks like the snapshot format changed, you might want to run `npm run test: new` or `npm run test: update` but that's about it.
I am thinking of adding a "Common failures and debugging / resolution" section to the README. Let's see when I get the time to do that.